addons/sourcemod/scripting/vsh/bosses/boss_uberranger.sp

```diff
@@ -0,0 +1,414 @@
+		259: Deals 3x falling damage to the player you land on
+		*/
+		
+		//We're selecting their color from a preset list
+		
+		//Checking if the list is there at all
+		if (g_aUberRangerColorList == null || g_aUberRangerColorList.Length <= 0)
+			UberRangerResetColorList();
+		
+		int iColor[3];
+			
+		g_aUberRangerColorList.GetArray(0, iColor);
+		SetEntityRenderColor(this.iClient, iColor[0], iColor[1], iColor[2], 255);
+		g_aUberRangerColorList.Erase(0);
+		
+		int iWearable = -1;
+		
+		//Another interesting thing: SetEntityRenderColor applies a color on top of the color it's already set for paintable models
+		//So for the color of the cosmetics to match the color of the Medic, we paint them white via an attribute then apply the random color
+		char sWhitePaint[16];
+		Format(sWhitePaint, sizeof(sWhitePaint), "142 ; 15132390");
+		
+		iWearable = this.CallFunction("CreateWeapon", 50, "tf_wearable", GetRandomInt(1, 100), TFQual_Normal, sWhitePaint);		//Prussian Pickelhaube
+		if (iWearable > MaxClients)
+		{
+			SetEntProp(iWearable, Prop_Send, "m_nModelIndexOverrides", g_iUberRangerPrussianPickelhaube);
+			SetEntityRenderColor(iWearable, iColor[0], iColor[1], iColor[2], 255);
+		}
+		
+		iWearable = this.CallFunction("CreateWeapon", 315, "tf_wearable", GetRandomInt(1, 100), TFQual_Normal, sWhitePaint);	//Blighted Beak
+		if (iWearable > MaxClients)
+		{
+			SetEntProp(iWearable, Prop_Send, "m_nModelIndexOverrides", g_iUberRangerBlightedBeak);
+			SetEntityRenderColor(iWearable, iColor[0], iColor[1], iColor[2], 255);
+		}
+	}
+	
+	public void GetModel(char[] sModel, int length)
+	{
+		strcopy(sModel, length, RANGER_MODEL);
+	}
+	
+	public void Destroy()
+	{
+		SetEntityRenderColor(this.iClient, 255, 255, 255, 255);
+	}
+};
+
+public void UberRangerResetColorList()
+{
+	if (g_aUberRangerColorList == null)
+		g_aUberRangerColorList = new ArrayList(3);
+	else
+		g_aUberRangerColorList.Clear();
+	
+	//TF2 colors
+	g_aUberRangerColorList.PushArray({ 125, 64, 113 }); 	// A Deep Commitment to Purple
+	g_aUberRangerColorList.PushArray({ 20, 20, 20 }); 		// A Distinctive Lack of Hue
+	g_aUberRangerColorList.PushArray({ 126, 126, 126 }); 	// Aged Mustache Grey
+	g_aUberRangerColorList.PushArray({ 40, 57, 77 }); 		// An Air of Debonair (BLU)
+	g_aUberRangerColorList.PushArray({ 231, 181, 59 }); 	// Australium Gold
+	g_aUberRangerColorList.PushArray({ 233, 150, 122 }); 	// Dark Salmon Injustice
+	g_aUberRangerColorList.PushArray({ 207, 115, 54 }); 	// Mann Co. Orange
+	g_aUberRangerColorList.PushArray({ 255, 105, 180 }); 	// Pink as Hell
+	g_aUberRangerColorList.PushArray({ 105, 77, 58 }); 		// Radigan Conagher Brown
+	g_aUberRangerColorList.PushArray({ 88, 133, 162 }); 	// Team Spirit (BLU)
+	g_aUberRangerColorList.PushArray({ 184, 59, 59 }); 		// Team Spirit (RED)
+	g_aUberRangerColorList.PushArray({ 50, 205, 50 }); 		// The Bitter Taste of Defeat and Lime
+	g_aUberRangerColorList.PushArray({ 66, 79, 59 }); 		// Zepheniah's Greed
```

Comment:
In `tf_econ_data.inc`, there `TF2Econ_GetPaintKitDefinitionList` and `TF2Econ_GetPaintKitDefinitionAddress` native for list of paint stuffs. I dunno how useful those are, or if even possible to use automated instead of listing here. Not absolute needed to use those anyway.

Reply:
Probably best to keep it hand-picked, I had to remove a lot of them for being lookalikes